4_advanced_usage/35_directive_ordering.txt

```diff
@@ -2,16 +2,76 @@
 
 Configuration in Rudder are based on desired states, describing the expected state of the system. However, there are cases where having order is desirable (like ensuring that a JVM is present before deploying an Application server, or ensuring a user is present before setting it sudoers), even if it will converge over the course of several agent runs.
 
-When policies are generated for each nodes, all Directives based on a same Technique are merged together, preventing a complete ordering of all Directives. However, Rudder uses a best-effort method for ordering Directives, based on alphanumeric ordering.
+In Rudder, there is two separated ways to order things, depending about what is the resulting policy. So, before that, we need to explain how Policies are generated on the
+agent. 
 
-Rudder tries to order first on Rules name, then on Directives name.
+==== Policy generation and Directive merge
 
-Example:
+In Rudder, Policies are generated from Directives, but several Directives based on the same Technique always lead to *one* Policy on the agent. 
+For unique (non multi-instance) Technique, the most propritary is selected. For multi-instance Technique, the different Directive values are *merged* 
+into one Policy after having been sorted. 
+
+.Separated Policy Generation in Rudder 4.3
+[TIP]
+=====
+ In Rudder 4.3, that limitation is lifted and Technique can be made to generate ONE Policy for each Directive. That capacity is controled by the 
+ `POLICYGENERATION` tag, where the value `merged` is the pre-4.3 default behavior, and values `separated` or `separated-with-param` lead to one Policy per Directive. 
+
+ See https://www.rudder-project.org/redmine/issues/10625[Don't merge directive from same technique on generation] for more information.
+=====
+
+
+==== Sorting Directives based on the *same* Technique
+
+For Directive based on the same Technique, the sort order is based on the *Priority* value of the Directive. Between two Directive, the one with the highest *Priority*
+is the first:
+
+- for a *non* multi-instance Technique, it means that it is the one that is chosen as the Policy, 
```

Comment:
it means that it is there is only one that is chosen in the resulting Policies (the others are discared),